### PR TITLE
Sentences List Screen Implementation

### PR DESCRIPTION
**Pull request by mkenlo**
In this PR, I built the app Home Page which displays a list of Sentences fetched for the API.

### CODE REVIEW

**Review by evindj — APPROVED** [review]
Good job!

File: lib/services/sentence_service.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import 'package:http/http.dart' as http;
+
+import '../models/sentence_model.dart';
+import 'settings.dart';
+
+
+const url = '$apiAuthorityUrl/sentences';
+
+Future<List<Sentence>> fetchSentences() async {
+  String filter = "language=francais";
```
Comment (evindj) [inline lib/services/sentence_service.dart:12]: Should not filter be const?

File: lib/services/sentence_service.dart [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  final response = await http.get('$url?$filter');
+
+  if (response.statusCode == 200) {
+    final jsonResponse = json.decode(response.body);
+    return Sentence.asListFromJson(jsonResponse["results"]);
+  } else {
+    throw Exception('Failed to load Data');
```
Comment (evindj) [inline lib/services/sentence_service.dart:19]: be a little more specific on the error message?

**Review by evindj — CHANGES_REQUESTED** [review]

File: lib/screens/sentence_item.dart [hunk 3]
```diff
@@ -1 +1,27 @@
...
+class SentenceItem extends StatelessWidget {
+  final Sentence _phrase;
+  final int _index;
+
+  SentenceItem(this._phrase, this._index);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: EdgeInsets.all(16.0),
```
Comment (evindj) [inline lib/screens/sentence_item.dart]: Is there some good practice for magic numbers?

**Review by evindj — APPROVED** [review]

File: lib/main.dart [hunk 4]
```diff
@@ -22,15 +24,16 @@ class MyApp extends StatelessWidget {
...
+    /*return Scaffold(
       appBar: AppBar(
         title: Text("Lingua"),
       ),
       body: Padding(
         padding: EdgeInsets.all(16.0),
         child: Text("Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."),
       )
-    );
+    );*/
```
Comment (evindj) [inline lib/main.dart]: Why is the code commented here? Please no commented code on production.

File: lib/screens/sentence_list_screen.dart [hunk 5]
```diff
@@ -1 +1,75 @@
...
+      )
+    );
+  }
+
+  Widget _loadSentencesWidget() {
+    return FutureBuilder(
+        future: fetchSentences(),
+        builder: (context, snapshot) {
+          if (snapshot.connectionState == ConnectionState.none) {
+            return _errorWidget("No connection to API");
```
Comment (evindj) [inline lib/screens/sentence_list_screen.dart]: can we be more specific about this? is it a server error or just that the user does not have internet connection?

File: lib/screens/sentence_list_screen.dart [hunk 6]
```diff
@@ -1 +1,75 @@
...
+          }
+        });
+  }
+
+  Widget _buildListWidget(data) {
+    return ListView.builder(
+        itemCount: data.length,
+        itemBuilder: (context, index) {
+          Sentence item = data[index];
+          return SentenceItem(item, index+1);
```
Comment (evindj) [inline lib/screens/sentence_list_screen.dart:43]: why ```index + 1```

File: lib/screens/sentence_item.dart [hunk 7]
```diff
@@ -8,20 +8,36 @@ class SentenceItem extends StatelessWidget {
 
   SentenceItem(this._phrase, this._index);
 
+  Color _tileBackgroundColor(BuildContext context) {
+    if (_index % 2 == 0) return Theme.of(context).primaryColorLight;
+    return Color.fromRGBO(1, 1, 1, 0.0);
```
Comment (evindj) [inline lib/screens/sentence_item.dart:13]: is it possible to express these without magic numbers?

**Review by evindj — APPROVED** [review]

**Review by evindj — APPROVED** [review]

**Review by evindj — APPROVED** [review]
The title of this diff does not seem to match the content.

File: lib/screens/sentence_item.dart (hunk 3 above)
Comment (mkenlo) [reply lib/screens/sentence_item.dart]: Will clean all of this later.

File: lib/main.dart (hunk 4 above)
Comment (mkenlo) [reply lib/main.dart]: Oups! Just forgot to clean it

File: lib/screens/sentence_list_screen.dart (hunk 6 above)
Comment (mkenlo) [reply lib/screens/sentence_list_screen.dart:43]: Because index starts at 0 in arrays and lists

File: lib/screens/sentence_item.dart (hunk 7 above)
Comment (mkenlo) [reply lib/screens/sentence_item.dart:13]: Will remove magic number later. Wanted something quick for the moment

File: lib/services/sentence_service.dart (hunk 1 above)
Comment (mkenlo) [reply lib/services/sentence_service.dart:12]: I don't see the need to make the variable a const

File: lib/services/sentence_service.dart (hunk 2 above)
Comment (mkenlo) [reply lib/services/sentence_service.dart:19]: How specific? Do you have examples?

**Pull request merged**